frontend/ui/widget/radiobutton.lua

```diff
@@ -0,0 +1,167 @@
+    self:update()
+    self.dimen = self.frame:getSize()
+    if Device:isTouchDevice() then
+        self.ges_events = {
+            TapCheckButton = {
+                GestureRange:new{
+                    ges = "tap",
+                    range = self.dimen,
+                },
+                doc = "Tap Button",
+            },
+            HoldCheckButton = {
+                GestureRange:new{
+                    ges = "hold",
+                    range = self.dimen,
+                },
+                doc = "Hold Button",
+            }
+        }
+    end
+end
+
+function RadioButton:update()
+    if self[1] then
+        self[1]:free()
+    end
+    self.frame = FrameContainer:new{
+        margin = self.margin,
+        bordersize = self.bordersize,
+        background = self.background,
+        radius = self.radius,
+        padding = self.padding,
+        LeftContainer:new{
+            dimen = Geom:new{
+                w = self.width,
+                h = self._widget_size.h
+            },
+            self._radio_button,
+        }
+    }
+    self[1] = self.frame
+end
+
+function RadioButton:onFocus()
+    self.frame.invert = true
+    return true
+end
+
+function RadioButton:onUnfocus()
+    self.frame.invert = false
+    return true
+end
+
+function RadioButton:onTapCheckButton()
+    if self.enabled and self.callback then
+        if G_reader_settings:isFalse("flash_ui") then
+            self.callback()
+        else
+            UIManager:scheduleIn(0.0, function()
+                self.invert = true
+                UIManager:setDirty(self.show_parent, function()
+                    return "ui", self.dimen
+                end)
+            end)
+            UIManager:scheduleIn(0.1, function()
+                self.callback()
+                self.invert = false
+                UIManager:setDirty(self.show_parent, function()
+                    return "ui", self.dimen
+                end)
+            end)
+        end
+    elseif self.tap_input then
+        self:onInput(self.tap_input)
+    elseif type(self.tap_input_func) == "function" then
+        self:onInput(self.tap_input_func())
+    end
+    return true
+end
+
+function RadioButton:onHoldCheckButton()
+    if self.enabled and self.hold_callback then
+        self.hold_callback()
+    elseif self.hold_input then
+        self:onInput(self.hold_input)
+    elseif type(self.hold_input_func) == "function" then
+        self:onInput(self.hold_input_func())
+    end
+    return true
+end
+
+function RadioButton:check(callback)
+    self._radio_button = self._checked_widget
+    self.checked = true
+    self:update()
+    UIManager:setDirty(self.parent, function()
+        return "partial", self.dimen
+    end)
+end
+
+function RadioButton:unCheck()
+    self._radio_button = self._unchecked_widget
+    self.checked = false
+    self:update()
+    UIManager:setDirty(self.parent, function()
+        return "partial", self.dimen
```

Comment:
You could use `"ui"` for such a small widget (thinking about Kindle users with REAGL where `"partial"` is really a fast but noticable "full")